src/interface/pipe.rs

```diff
@@ -89,60 +168,127 @@ impl EmulatedPipe {
             return false;
         }
     }
+
+    /// # Description
+    /// Checks if pipe is currently ready for writing, used by select/poll etc
+    ///
+    /// # Returns
+    ///
+    /// True if descriptor is ready for writing, false if it will block
+    ///
     pub fn check_select_write(&self) -> bool {
         let write_end = self.write_end.lock();
         let pipe_space = write_end.remaining();
 
-        return pipe_space != 0;
+        // Linux considers a pipe writeable if there is at least PAGE_SIZE (PIPE_BUF) remaining space (4096 bytes)
+        return (self.size - pipe_space) > PAGE_SIZE;
```

Comment:
Looks like Linux also considers a pipe to be writable if there is no readers

Reply:
Really? thats interesting. Can you link me to that?

I guess its for software that is selecting on a closed pipe to error out as quickly as possible.

Reply:
Its just inside the pipe_writable() under pipe.c, I'll paste it here:
static inline bool pipe_writable(const struct pipe_inode_info *pipe)
{
	unsigned int head = READ_ONCE(pipe->head);
	unsigned int tail = READ_ONCE(pipe->tail);
	unsigned int max_usage = READ_ONCE(pipe->max_usage);

	return !pipe_full(head, tail, max_usage) ||
		!READ_ONCE(pipe->readers);
}

Reply:
thanks!